### PR TITLE
Update state markup for VAOS addresses

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This PR:
- Updates our State component with recommended markup from our VSP to avoid an aXe error while still providing the full state info to screen reader users
- Updates the tests to deal with the new text

## Testing done
Local and unit testing. Verified there's no aXe warning on details page

## Screenshots
Screen Shot 2021-06-23 at 3 05 00 PM

## Acceptance criteria
- [ ] State text works correctly

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.cancel.unit.spec.jsx [hunk 1]
```diff
@@ -404,7 +404,7 @@ describe('VAOS <AppointmentsPage> cancellation:', () => {
     expect(modal).to.contain.text('Something went wrong');
     expect(modal).to.contain.text('Cheyenne VA Medical Center');
     expect(modal).to.contain.text('2360 East Pershing Boulevard');
-    expect(modal).to.contain.text('Cheyenne, WY 82001-5356');
+    expect(modal).to.contain.text('Cheyenne, WyomingWY 82001-5356');
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.cancel.unit.spec.jsx:407]: I dislike this, and I'm open to other ideas about how to verify these addresses. Both the full name and the abbreviation are part of `textContent` (they're shown/hidden with css and aria attributes). JSDom doesn't support innerText, which in theory might only show what's visible. You could break this into separate expects, but that just hides the weird way the text content looks.

File: src/applications/vaos/components/State.jsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 
-const stateNames = {
+export const stateNames = {
```
Comment (jbalboni) [inline src/applications/vaos/components/State.jsx:3]: Using these state names in a test

File: src/applications/vaos/components/State.jsx [hunk 3]
```diff
@@ -10,8 +10,8 @@ const stateNames = {
   CO: 'Colorado',
   CT: 'Connecticut',
   DE: 'Delaware',
-  DC: 'District Of Columbia',
-  FM: 'Federated States Of Micronesia',
+  DC: 'District of Columbia',
```
Comment (jbalboni) [inline src/applications/vaos/components/State.jsx:13]: Not sure why these were capitalized before.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.cancel.unit.spec.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.cancel.unit.spec.jsx:407]: Just curious, but what changed that the original test is not working?

Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.cancel.unit.spec.jsx:407]: Agreed! Nothing comes to mind other than using regex which just makes things worse.

**Review by vbahinwillit — APPROVED** [review]

**Pull request merged**